packages/babel-types/src/clone/cloneNode.js

```diff
@@ -42,6 +42,8 @@ export default function cloneNode<T: Object>(node: T, deep: boolean = true): T {
       newNode.typeAnnotation = deep
         ? cloneIfNodeOrArray(node.typeAnnotation, true)
         : node.typeAnnotation;
+
+      newNode.optional = node.optional;
```

Comment:
Shouldn't this be in a separate if check?

Reply:
I was under impression that it can only be marked as optional when there's a type annotation present. But you're right, looks like it's not necessary and `function f(foo?) {}` is still perfectly valid.

Will be fixed in a second